MedicalPoint/Services/MedicinesService.cs

```diff
@@ -48,6 +51,16 @@ public async Task<Medicine> Get(int medicineId, CancellationToken cancellationTo
             return result;
         }
 
+
+        public async Task<Medicine> Delete(int medicineId, CancellationToken cancellationToken = default)
+        {
+            var result = await _context.Medicines.Include(x => x.History).AsNoTracking().FirstOrDefaultAsync(x => x.Id == medicineId, cancellationToken);
+
+            _context.Medicines.Remove(result);
```

Comment:
don't remove the medicine from the database, change IsDeleted property to true & add row for this operation